lib/state_machine/state_machine.cpp

```diff
@@ -4,7 +4,9 @@
 
 namespace hyped::state_machine {
 
-StateMachine::StateMachine() : current_state_{State::kIdle}
+StateMachine::StateMachine(const TransitionTable &transition_table)
+    : current_state_{State::kIdle},
```

Comment:
This was my mistake a while ago, quick fix here
```suggestion
    : current_state_(State::kIdle),
```